src/v/pandaproxy/handlers.cc

```diff
@@ -81,6 +85,43 @@ get_topics_names(server::request_t rq, server::reply_t rp) {
       });
 }
 
+ss::future<server::reply_t>
+get_topics_records(server::request_t rq, server::reply_t rp) {
+    auto fmt = parse_serialization_format(rq.req->get_header("Accept"));
+    if (fmt == serialization_format::unsupported) {
+        rp.rep = unprocessable_entity("Unsupported serialization format");
+        return ss::make_ready_future<server::reply_t>(std::move(rp));
+    }
+
+    model::topic_partition tp{
+      model::topic(rq.req->param["topic_name"]),
+      model::partition_id{boost::lexical_cast<model::partition_id::type>(
+        rq.req->param["partition_id"])}};
+    model::offset offset{boost::lexical_cast<model::offset::type>(
+      rq.req->get_query_param("offset"))};
+    std::chrono::milliseconds timeout{
+      boost::lexical_cast<std::chrono::milliseconds::rep>(
```

Comment:
i guess lexical_cast here will throw if the url query parameters contain something like non-numeric values. does `get_topic_records` behave reasonable in this case?

Reply:
Yeah, we can probably do something nicer than:
```
{"message": "boost::wrapexcept<boost::bad_lexical_cast> (bad lexical cast: source type value could not be interpreted as target)", "code": 500}
```

Reply:
I think I'll raise a separate issue for this, it's a bit messy.

Reply:
https://github.com/vectorizedio/redpanda/issues/145